specs/plugin-extensibility.md

```diff
@@ -366,14 +366,15 @@ This interface targets plugins that in addition to signature generation want to
 1. Execute the plugin with `get-plugin-metadata` command
     1. If plugin supports capability `SIGNATURE_GENERATOR.ENVELOPE`
         1. Execute the plugin with `generate-envelope` command. Set `request.keyId` and the optional `request.pluginConfig` to corresponding values associated with signing key `keyName` in `config.json`. Set `request.payload` to base64 encoded the [Notary Project signature Payload](./signature-specification.md#payload), `request.payloadType` to `application/vnd.cncf.notary.payload.v1+json` and `request.signatureEnvelopeType` to a pre-defined type (`application/jose+json` for JWS).
-        2. `response.signatureEnvelope` contains the base64 encoded signature envelope, value of `response.signatureEnvelopeType` MUST match `request.signatureEnvelopeType`.
-        3. Validate the generated signature, return an error if of the checks fails.
+        2. If plugin supports timestamping under signing scheme [`notary.x509`](./signing-scheme.md/#notaryx509), the plugin MUST request and include TSA timestamp countersignature in the signature envelope at this step. Plugin MAY also allow users to specify their own TSA server URL via [pluginConfig](#plugin-configuration).
```

Comment:
How does `notation` know a plugin supports timestamping? or it is purely configured by `plugin-config` by individual plugins?

Reply:
I guess this question is repeated? This is `Signature Envelope Generator`, plugin has full control of the signature envelope including any timestamp countersignature in it. The plugin publish could choose to expose the TSA they used in their release doc for user references. They could also choose to take user specified `tsa url` from `--plugin-config`. Let plugins make the decisions is more flexible.